### ISSUE TITLE
Add debugging symbols for gpus.

### ISSUE THREAD

**Issue by marthaisabelhilton** [GooFit/GooFit#221]
Dear GooFit experts,
Is it possible to add debugging symbols for the gpu when using cuda-gdb?
I've tried adding `-G` to `CMAKE_CUDA_FLAGS` and I get the following error: 
`there was an error in verifying the lgenfe output`
Thanks,
Martha

**Comment by henryiii** [comment]
I *think* the following should work:

```bash
cmake <build_path> -DCMAKE_CUDA_FLAGS_DEBUG="-g -G" -DCMAKE_BUILD_TYPE=Debug
```

(compiles on CUDA 10.0, at least)

**Comment by henryiii** [comment]
(If it does work, we'll add it to the documentation and close the issue.)

**Comment by danielsibemol** [comment]
When I try this flag the compilation fails:
/usr/include/c++/4.8.2/cmath(605): error: calling a __host__ function("__builtin_isinf") from a __device__ function("std::isinf") is not allowed

/usr/include/c++/4.8.2/cmath(609): error: calling a __host__ function("__builtin_isinf") from a __device__ function("std::isinf") is not allowed

/usr/include/c++/4.8.2/cmath(623): error: calling a __host__ function("__builtin_isnan") from a __device__ function("std::isnan") is not allowed

/usr/include/c++/4.8.2/cmath(627): error: calling a __host__ function("__builtin_isnan") from a __device__ function("std::isnan") is not allowed

/usr/include/c++/4.8.2/cmath(659): error: calling a __host__ function("__builtin_signbit") from a __device__ function("std::signbit") is not allowed

/usr/include/c++/4.8.2/cmath(663): error: calling a __host__ function("__builtin_signbit") from a __device__ function("std::signbit") is not allowed

6 errors detected in the compilation of "/tmp/tmpxft_000053dc_00000000-7_MetricPointer.cpp1.ii".
gmake[2]: ** [src/PDFs/CMakeFiles/PDFCore.dir/MetricPointer.cu.o] Erro 2

This error message pops up for both src/PDFs/CMakeFiles/PDFCore.dir/MetricPointer.cu and src/PDFs/CMakeFiles/PDFCore.dir/utilities/VariableBinTransform1DPdf.cu. This error is a bit puzzling for me because neither of these use any kind of isnan or isinf.

**Comment by henryiii** [comment]
Update: Cuda 8 fails with the above error message, it seems that cmath can't be included in a debug build (GCC 4.8 on CentOS 7). Any other CUDA version (9.0, 9.2, 10.0) fails with: `Error: Internal Compiler Error (codegen): "there was an error in verifying the lgenfe output!"`.

**Comment by thboettc** [comment]
We successfully built with debug symbols during the hackathon, so I think we can close this. I also think the compiler error @henryiii mentions was fixed in https://github.com/GooFit/GooFit/pull/298.
